xdsl/dialects/x86/ops.py

```diff
@@ -320,6 +321,48 @@ class PushOp(ROperationSrc[GeneralRegisterType]):
     name = "x86.push"
 
 
+class ROperationSrcDst(Generic[R1InvT], SingleOperandInstruction):
+    """
+    A base class for x86 operations that have one register acting as both source and destination.
+    """
+
+    T = Annotated[GeneralRegisterType, ConstraintVar("T")]
+    source = operand_def(R1InvT)
+    destination = result_def(R1InvT)
+
+    def __init__(
+        self,
+        source: Operation | SSAValue | None = None,
+        *,
+        comment: str | StringAttr | None = None,
+        destination: R1InvT | None = None,
+    ):
+        if isinstance(comment, str):
+            comment = StringAttr(comment)
+
+        super().__init__(
+            operands=[source],
+            attributes={
+                "comment": comment,
+            },
+            result_types=[destination],
+        )
+
+    def assembly_line_args(self) -> tuple[AssemblyInstructionArg | None, ...]:
+        return (self.destination,) if self.destination else (self.source,)
```

Comment:
Both of these are guaranteed not to be `None`, might as well just pick one

Reply:
artifact from the previous version, my bad